src/MassTransit/Configuration/DefaultEndpointNameFormatter.cs

```diff
@@ -199,8 +213,10 @@ string GetActivityName<T>()
 
         string FormatName(Type type)
         {
+            var joinSeparator = string.IsNullOrEmpty(_joinSeparator) ? string.Empty : _joinSeparator;
+
             var name = _includeNamespace
-                ? string.Join("", TypeCache.GetShortName(type).Split(_removeChars))
+                ? string.Join(joinSeparator, TypeCache.GetShortName(type).Split(_removeChars))
```

Comment:
doesn't this need a `joinSeparator ?? ""` to avoid a NRE?

Reply:
Oh, I see, I'd just use the shorter syntax, versus the string.Empty code above.

Reply:
Alright done